src/main.cpp

```diff
@@ -1337,12 +1359,15 @@ bool ContextualCheckTransaction(
         librustzcash_sapling_verification_ctx_free(ctx);
     }
 
+    // Queue Orchard signatures to be batch-validated.
+    tx.GetOrchardBundle().QueueSignatureValidation(orchardAuth, dataToBeSigned);
```

Comment:
For Orchard we no longer revalidate using the previous epoch's branch ID, is that correct (because that's more difficult with / would nullify some of the advantages of batch validation)?

Reply:
We don't need to. The purpose of that double-validation was because we had no way of knowing what consensus branch ID the transaction creator used. v5 transactions store the consensus branch ID directly, so if a transaction was signed with the previous epoch's branch ID, we'll have rejected it in `ContextualCheckTransaction` before we reach this point.

Reply:
Cool, and the error message includes both the encoded and expected branch IDs. :+1: 

https://github.com/zcash/zcash/blob/6cd5b8792bbe0a3cc1064e1884914415f0c6e7d7/src/main.cpp#L1123-L1131